WooCommerce/Classes/Authentication/Store Creation/StoreCreationCoordinator.swift

```diff
@@ -68,8 +64,8 @@ final class StoreCreationCoordinator: Coordinator {
         self.featureFlagService = featureFlagService
 
         Task { @MainActor in
-            if let iapManager {
-                self.iapManager = iapManager
+            if let purchasesManager {
+                self.purchasesManager = purchasesManager
             }
         }
```

Comment:
❓ Just curious - why does this have to be called async?

Reply:
It's because the IAP implementation `InAppPurchasesForWPComPlansManager` is a `MainActor`, and thus this `purchasesManager` property has to also be `MainActor`. Because the property is a `MainActor`, setting/mutating it requires it to be from a `MainActor` context. We could make the `StoreCreationCoordinator` initializer to be `MainActor`, but that introduces a lot of changes upstream so I thought it's easier to set it this way.

Without the Task/MainActor context, an error is thrown:
```
Main actor-isolated property 'purchasesManager' can not be mutated from a non-isolated context
```